### PR TITLE
Update Java ITR and Jenkins plugin documentation

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### What does this PR do? What is the motivation?

This PR contains the following updates:
- Karate is added to the list of Java testing frameworks for which ITR is supported
- In the Jenkins plugin documentation the list of minimum required Git info fields is specified: repo URL, branch, commit SHA and author email

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Review by anmarchenko — APPROVED** [review]

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 1]
```diff
@@ -1218,7 +1218,13 @@ pipeline {
 }
 {{< /code-block >}}
 
-## Propagate Git information from SCM
+## Propagate Git information
+
+Some features of the Datadog Jenkins plugin need to know Git information associated with the Jenkins builds in order to work correctly.
```
Comment (brett0000FF) [inline content/en/continuous_integration/pipelines/jenkins.md:1223]:
```suggestion
Some features of the Datadog Jenkins plugin need Git information associated with the Jenkins builds to work correctly.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 2]
```diff
@@ -1218,7 +1218,13 @@ pipeline {
...
+The minimum required Git info for a build is repository URL, branch, commit SHA and commit author email.
```
Comment (brett0000FF) [inline content/en/continuous_integration/pipelines/jenkins.md:1224]:
```suggestion
The minimum required Git info for a build is repository URL, branch, commit SHA, and commit author email.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 3]
```diff
@@ -1218,7 +1218,13 @@ pipeline {
...
+These can be determined by the plugin automatically, propagated from SCM, provided manually via environment variables, or obtained by combining these approaches.
```
Comment (brett0000FF) [inline content/en/continuous_integration/pipelines/jenkins.md:1225]:
```suggestion
This information can be determined by the plugin automatically, propagated from SCM, provided manually with environment variables, or obtained by combining these approaches.
```

**Review by brett0000FF — APPROVED** [review]
Thanks! Looks good. I just suggested a few style tweaks. After you take a look, let me know, and I can merge this. 👍

**Comment by nikita-tkachenko-datadog** [comment]
> Thanks! Looks good. I just suggested a few style tweaks. After you take a look, let me know, and I can merge this. 👍

Thank you, applied the suggestions. All looks good from my perspective to proceed with the merge

**Review by brett0000FF — APPROVED** [review]

**Pull request merged**